internal/flybarman/node.go

```diff
@@ -0,0 +1,194 @@
+package flybarman
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	"github.com/fly-apps/postgres-flex/internal/flypg"
+	"github.com/fly-apps/postgres-flex/internal/flypg/admin"
+)
+
+var dataDir = "/data"
+var barmanConfigFile = dataDir + "/barman.conf"
+var barmanCronFile = dataDir + "/barman.cron"
+var globalBarmanConfigFile = "/etc/barman.conf"
+
+type Node struct {
+	AppName       string
+	PrivateIP     string
+	PrimaryRegion string
+	DataDir       string
+	Port          int
+
+	BarmanHome         string
+	LogFile            string
+	PasswordConfigPath string
+
+	SUCredentials       admin.Credential
+	OperatorCredentials admin.Credential
+	ReplCredentials     admin.Credential
+}
+
+func NewNode() (*Node, error) {
+	node := &Node{
+		AppName:            "local",
+		BarmanHome:         dataDir + "/barman.d",
+		LogFile:            dataDir + "/barman.log",
+		PasswordConfigPath: "/root/.pgpass",
+	}
+
+	if appName := os.Getenv("FLY_APP_NAME"); appName != "" {
+		node.AppName = appName
+	}
+
+	// Internal user
+	node.SUCredentials = admin.Credential{
+		Username: "flypgadmin",
+		Password: os.Getenv("SU_PASSWORD"),
+	}
+
+	// Postgres user
+	node.OperatorCredentials = admin.Credential{
+		Username: "postgres",
+		Password: os.Getenv("OPERATOR_PASSWORD"),
+	}
+
+	// Repmgr user
+	node.ReplCredentials = admin.Credential{
+		Username: "repmgr",
+		Password: os.Getenv("REPL_PASSWORD"),
+	}
+
+	return node, nil
+}
+
+func (n *Node) Init(_ context.Context) error {
+	err := flypg.WriteSSHKey()
+	if err != nil {
+		return fmt.Errorf("failed write ssh keys: %s", err)
+	}
+
+	if _, err := os.Stat(barmanConfigFile); os.IsNotExist(err) {
+		barmanConfigFileContent := fmt.Sprintf(`[barman]
+barman_user = root
+barman_home = /data/barman.d
+log_level = info
+log_file = /data/barman.log
+
+[pg]
+description =  "Fly.io Postgres Cluster"
+conninfo = host=%s.internal user=repmgr dbname=postgres
+streaming_conninfo = host=%s.internal user=repmgr dbname=postgres
+backup_method = postgres
+streaming_archiver = on
+slot_name = barman
+create_slot = auto
+retention_policy_mode = auto
```

Comment:
We may want to rethink this approach a bit as some of these settings users will need to tune.

Reply:
@davissp14 these settings are written once an then the user can update as needed. Does that sound ok?

Reply:
What's the process for updating those settings?

Reply:
Oh nvm, first time I read it I thought the settings would be overwritten on restart.  This should be fine for now.